src/backend/distributed/executor/multi_utility.c

```diff
@@ -290,6 +291,19 @@ multi_ProcessUtility(Node *parsetree,
 		}
 	}
 
+	/* handle DDL on the worker node */
+	if (!schemaNode)
```

Comment:
Hmz, there's a slight issue here. What if the user typed a DDL command on the schema node? We would normally want to error out saying that it should go to the master. In the future, we might want to forward the command to the schema node. 

I guess we could:
- sneakily allow foreign key DDL to be executed
- always skip validation of foreign key if enable_ddl_propagation is off
- introduce a new GUC for metadata propagation
- go through a UDF

I kind of like option 2 since it hurts the least.